docs/src/references/synchronize-daemons.md

```diff
@@ -0,0 +1,116 @@
+ So, better to check the status of the bridge before sending the request to the c8y cloud.
+
+3. Some of the daemons are dependent on the `tedge-mapper-c8y`(cloud mapper) file system to create the supported operations file.
+
+In the existing solution, the thin-edge daemon interdependency was solved using a workaround by
+creating the `persistent session` with the broker when the `daemon is installed`. 
+For example, when the tedge-agent is installed, it will create a session with the MQTT broker. So, when the tedge-agent is down the broker
+will buffer the messages and delivers once it comes up.
+
+The problem with the existing workaround is
+- The MQTT broker has to be up and running while installing the thin-edge components to create the session
+- If in case the init session fails, then thin-edge daemons have to be started in a strict order,
+ else the first message will be lost. i.e tedge-agent has to be started before the tedge-mapper-c8y
+- Also, if the init session is done twice, then also the messages will be lost
+
+
+# Proposed solutions and their Pros and Cons
+
+## Proposal 1: Use the `health check` mechanism
+
+The mapper uses the health-check mechanism to detect the health of another daemon and perform all the cloud-specific
+ actions for that daemon like updating the supported operations etc.
+In this, the `daemon` sends the health status `up` message once it's completely up. Then the `mapper` picks up this message and creates the required operation files and also sends the supported operation to the cloud.
+Also, now the mapper can start interacting with the daemon.
+
+Example 1,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/health/#` topic
+- When tedge-agent is started, it will publish health status as `up` on to `tedge/health/tedge-agent` topic
+- Now the mapper picks up this message and creates the supported operations file (c8y_SoftwareUpdate, etc) in `/etc/tedge/operations/c8y` with their content. 
+- Mapper sends the updated supported operations list to the c8y cloud.
+- Also, now that the tedge-agent is up and running, the tedge-mapper-c8y will publish the request to get the
+ software list on `tedge/commands/req/software/list`.
+- Now the tedge-agent processes this request and sends the software list to the mapper, once the mapper gets the response it will forward it to the c8y cloud.
+
+Example 2,
+- When tedge-mapper-c8y is started, it will subscribe to `tedge/health/#` topic
+- When c8y-log-plugin is started, it will publish health status as `up` on to `tedge/health/c8y-log-plugin` topic
+- Now the mapper picks up this message and creates the c8y_LogfileRequest operation file in `/etc/tedge/operations/c8y`.
+- Also the mapper creates the `c8y-log-plugin.toml` file, which contains the `supported log types` in `/etc/tedge/c8y`.
+- Mapper sends the updated supported operations list to the c8y cloud. Also, it sends the supported `log types`(118,software-management) to the c8y cloud.
```

Comment:
> Because when the log-plugin restarts, this will emit the health status again,

This is what I meant by flaky: depending a plugin is running or not the operation will available or not in the cloud.

I think that:
- A capability must be displayed as available to the user when the associated plugin is installed.
- If the plugin is not running, stopped or broken, then the capability is not operational but still available (i.e. listed among the things the device is supposed ready to do).